From fd4ca689cb0711f1b4c37405e0086f7e1bd5471d Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@cloudera.com>
Date: Mon, 19 Mar 2012 20:23:27 -0700
Subject: [PATCH 1124/1357] MR1: MAPREDUCE-4036 Streaming TestUlimit fails on CentOS 6 (MR1) (tucu)

  Reason: CentOS 6 JVM has higher minimum memory requirements
  Author: Alejandro Abdelnur
  Ref: CDH-4893
---
 .../org/apache/hadoop/streaming/TestUlimit.java    |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/contrib/streaming/src/test/org/apache/hadoop/streaming/TestUlimit.java b/src/contrib/streaming/src/test/org/apache/hadoop/streaming/TestUlimit.java
index c555e2e..b14ce30 100644
--- a/src/contrib/streaming/src/test/org/apache/hadoop/streaming/TestUlimit.java
+++ b/src/contrib/streaming/src/test/org/apache/hadoop/streaming/TestUlimit.java
@@ -46,7 +46,7 @@ public class TestUlimit extends TestCase {
   MiniDFSCluster dfs = null;
   MiniMRCluster mr = null;
   FileSystem fs = null;
-  private static String SET_MEMORY_LIMIT = "786432"; // 768MB
+  private static String SET_MEMORY_LIMIT = "1786432"; // 1768MB
 
   String[] genArgs(String memLimit) {
     return new String[] {
@@ -57,6 +57,8 @@ public class TestUlimit extends TestCase {
       "-numReduceTasks", "0",
       "-jobconf", "mapred.map.tasks=1",
       "-jobconf", JobConf.MAPRED_MAP_TASK_ULIMIT + "=" + memLimit,
+      "-jobconf", "mapred.child.java.opts=-Xmx512m",
+      "-jobconf", "mapred.child.env=MALLOC_ARENA_MAX=1",
       "-jobconf", "mapred.job.tracker=" + "localhost:" +
                                            mr.getJobTrackerPort(),
       "-jobconf", "fs.default.name=" + "hdfs://localhost:" 
-- 
1.7.0.4

